Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy+DisambiguatedPaths.swift

```diff
@@ -182,7 +190,24 @@ extension PathHierarchy.DisambiguationContainer {
     
     static func disambiguatedValues(
         for elements: some Sequence<Element>,
-        includeLanguage: Bool = false
+        includeLanguage: Bool = false,
+        allowAdvancedDisambiguation: Bool = true
+    ) -> [(value: PathHierarchy.Node, disambiguation: Disambiguation)] {
+        var collisions = _disambiguatedValues(for: elements, includeLanguage: includeLanguage, allowAdvancedDisambiguation: allowAdvancedDisambiguation)
+        
+        // If all but one of the collisions are disfavored, remove the disambiguation for the only favored element.
+        if let onlyFavoredElementIndex = collisions.firstIndex(where: { !$0.value.specialBehaviors.contains(.disfavorInLinkCollision) }),
```

Comment:
Oh yeah, I'm accessing the option set directly. I can create convenience accessors for these.

Reply:
> Can you use the `singleMatch` extension to Sequence here?

No, because the collision elements are value types and I want to in-place modify the array of collisions.

Reply:
Convenience accessors added in https://github.com/swiftlang/swift-docc/pull/643/commits/216a0383e5a0a1d6a4f6cca067439a4ed28e0349

`onlyIndex(where:)` utility extracted in https://github.com/swiftlang/swift-docc/pull/643/commits/d2045a8f94fb96245a8d451d8479903c0c241acc